docs/proposals/20231122-automate-aks-features.md

```diff
@@ -0,0 +1,397 @@
+suitable to implement in CAPZ and users are interested in having all AKS features available to them from
+Cluster API.
+
+When gaps exist in the set of features available in AKS and what CAPZ offers, users of other infrastructure
+management solutions may not be able to adopt CAPZ. If all AKS features could be used from CAPZ, this would
+not be an issue.
+
+The AKS feature set changes rapidly alongside CAPZ's users' desire to utilize those features. Because making
+new AKS features available in CAPZ requires a considerable amount of mechanical, manual effort to implement,
+review, and test, requests for new AKS features account for a large portion of the cost to maintain CAPZ.
+
+Another long-standing feature request in CAPZ is the ability to adopt existing AKS clusters into management by
+Cluster API (https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1173). Making the entire AKS
+API surface area available from CAPZ is required to enable this so existing clusters' full representations can
+be reflected in Cluster API.
+
+### Goals
+
+- Narrow the gap between the sets of features offered by AKS and CAPZ.
+- Reduce the maintenance cost of making new AKS features available from CAPZ.
+- Preserve the behavior of existing CAPZ AKS definitions while allowing users to utilize the new API pattern
+  iteratively to use new features not currently implemented in CAPZ.
+
+### Non-Goals/Future Work
+
+- Automate the features available from other Azure services besides AKS.
+- Automatically modify AKS cluster definitions to transparently enable new AKS features.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a managed cluster user, I want to be able to use all available AKS features natively from CAPZ so that I
+can more consistently manage my CAPZ AKS clusters that use advanced or niche features more quickly than having
+to wait for each of them to be implemented in CAPZ.
+
+#### Story 2
+
+As an AKS user looking to adopt Cluster API over an existing infrastructure management solution, I want to be
+able to use all AKS features natively from CAPZ so that I can adopt Cluster API with the confidence that all
+the AKS features I currently utilize are still supported.
+
+#### Story 3
+
+As a CAPZ developer, I want to be able to make new AKS features available from CAPZ more easily in order to
+meet user demand.
+
+### API Design Options
+
+There are a few different ways the entire AKS API surface area could be exposed from the CAPZ API. The
+following options all rely on ASO's ManagedCluster and ManagedClustersAgentPool resources to define the full
+AKS API, roughly organized in order of increasing responsibility for CAPZ. The examples below use
+AzureManagedControlPlane and ManagedCluster to help illustrate, but all of the same ideas should also apply to
+AzureManagedMachinePool and ManagedClustersAgentPool.
+
+#### Option 1: CAPZ resource references an existing ASO resource
```

Comment:
Here's a potential representation of what option 1 would look like in practice:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4486/files

(might need to manually expand diff)

The assumption there is that we would cut a new v2beta1 api to support the evolution of the CAPZ AKS resources.